Fixes dispose being called twice on TestFixtures with TestCases

rprouse opened the pull request:
Fixes #801 by ensuring that the `Test` is a `TestFixture` before disposing it because the `context.TestObject` is set for the `ParameterizedMethodSuite` for the `TestCases`. We don't want children of a `TestFixture` disposing the parent.

rprouse commented:
Failed Travis run is an unrelated random error. I don't have much faith in Travis with all these errors. Or should my lack of faith be directed at Mono?

```
1) Failed : NUnit.Framework.Internal.Execution.WorkItemQueueTests.StopQueue_WithWorkers
		  Worker thread 2 did not stop
		  Expected: False
		  But was:  True
		at NUnit.Framework.Internal.Execution.WorkItemQueueTests.StopQueue_WithWorkers () [0x00000] in <filename unknown>:0
```

CharliePoole commented:
I think the problem is that we have some rather non-robust tests that depend on timing. We kept playing with them till they passed on Windows but they still fail on Linux. Really, we should rethink the design of these tests.

CharliePoole commented on src/NUnitFramework/framework/Internal/Commands/OneTimeTearDownCommand.cs:
```diff
@@ -69,7 +69,7 @@ public override TestResult Execute(TestExecutionContext context)
 
                 IDisposable disposable = context.TestObject as IDisposable;
-                if (disposable != null)
+                if (disposable != null && Test is TestFixture)
```
What if it's a SetUpFixture? Ideally, we should have a common base class for all those tests that are represented by Types we must create. For now, I'd add a check for SetUpFixture and maybe a TODO comment.

rprouse commented:
Good catch @CharliePoole. We may as well do it right now. I added an `IDisposableFixture` interface to apply to fixtures that should have their attached class disposed. I applied it to `TestFixture` and `SetupFixture`.

CharliePoole commented:
I'm OK with merging this but I'm wondering about the introduction of the NUnit.Interfaces namespace, as opposed to the existing NUnit.Framework.Interfaces.

rprouse commented:
I will fix the namespace and then merge. I guess we don't have the default namespace set correctly on one of the projects? I will look.

rprouse commented:
The default namespace wasn't set correctly on the portable and the SilverLight projects. I fixed them and fixed the namespace for the interface. I will merge once CI passes.
